File: Client/Captions/CaptionsLib/source/fab/FABUtils.vb
'Filename: FABUtils.vb
'Version: 20121015

Imports ClipFlair.CaptionsLib.utils.DateTimeUtils

Namespace ClipFlair.CaptionsLib.FAB

  Public NotInheritable Class FABUtils

    Public Const FABtimeFormat As String = "HH:mm:ss:ff"
    Public Const SignificantDigits As Integer = 2

    Public Shared BaseTime As DateTime = DATETIMEZERO

    Public Shared Function SecondsToFABtime(ByVal seconds As Double) As String
      Return SecondsToDateTimeStr(seconds, BaseTime, FABtimeFormat, SignificantDigits)
    End Function

    Public Shared Function FABtimeToSeconds(ByVal FABTime As String) As Double
      Return TimeStrToSeconds(FABTime, BaseTime, FABtimeFormat, SignificantDigits)
    End Function

  End Class

End Namespace

File: Client/Captions/CaptionsLib/source/fab/FABUtils.cs
//Filename: FABUtils.cs
//Version: 20131105

using System;
using ClipFlair.CaptionsLib.Utils;

namespace ClipFlair.CaptionsLib.FAB
{

	public sealed class FABUtils
	{

		public const string FABtimeFormat = "HH:mm:ss:ff";

		public const int SignificantDigits = 2;

		public static DateTime BaseTime = DateTimeUtils.DATETIMEZERO;
		public static string SecondsToFABtime(double seconds)
		{
      return DateTimeUtils.SecondsToDateTimeStr(seconds, BaseTime, FABtimeFormat, SignificantDigits);
		}

		public static double FABtimeToSeconds(string FABTime)
		{
      return DateTimeUtils.TimeStrToSeconds(FABTime, BaseTime, FABtimeFormat, SignificantDigits);
		}

	}

}